libraries/engage/push-opt-in/subscriptions/optInTrigger.js

```diff
@@ -0,0 +1,112 @@
+import {
+  appConfig,
+} from '@shopgate/engage';
+import {
+  main$,
+  event,
+  appDidStart$,
+  getAppPermissions,
+  PERMISSION_ID_PUSH,
+  PERMISSION_STATUS_NOT_DETERMINED,
+} from '@shopgate/engage/core';
+import {
+  increaseAppStartCount,
+  resetAppStartCount,
+  increaseOrdersPlacedCount,
+  resetOrdersPlacedCount,
+  setLastPopupTimestamp,
+  increaseRejectionCount,
+} from '../action-creators';
+import {
+  PUSH_OPT_IN_OPT_IN_POSTPONED,
+} from '../constants';
+
+import { getPushOptInTriggerState } from '../selectors';
+import { showOptIn } from '../actions';
+
+const DAY_IN_MS = 1000 * 60 * 60 * 24;
+
+const increaseRejectionCount$ = main$.filter(
```

Comment:
like the `app-rating` feature i consider the current content of the `push-opt-in` feature as encapsulated from the rest of the app. for that reason there is not index file right now that can be used to import e.g. actions. interfering the logic inside the subscriptions could cause side effects, so if different logic would be needed (e.g. provided via an extension) we'd need to provide an option to turn off the default behaviour.

since `increaseRejectionCount$` is not supposed to be a public api, and additionally it's the only custom stream for that feature, it thought that it's ok to put it here. there are a couple of other examples in the PWA project where it's handled like that.

i implemented it like this to keep the reducer as dump as possible. all relevant business logic to maintain the different states is supposed to be implemented inside the subscriptions.